integrationtest/mapper/orm-spring/src/test/java/org/hibernate/search/integrationtest/spring/jta/TransactionTimeoutJtaAndSpringOutboxIT.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.spring.jta;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import javax.persistence.EntityManagerFactory;
+
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.resource.transaction.spi.TransactionCoordinatorBuilder;
+import org.hibernate.search.integrationtest.spring.jta.dao.SnertDAO;
+import org.hibernate.search.integrationtest.spring.jta.entity.Snert;
+import org.hibernate.search.integrationtest.spring.jta.listener.TimeoutFailureCollector;
+import org.hibernate.search.mapper.orm.session.impl.HibernateOrmSearchSessionHolder;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = JtaAndSpringOutboxApplicationConfiguration.class)
+@ActiveProfiles({ "jta", "outbox", "transaction-timeout" })
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class TransactionTimeoutJtaAndSpringOutboxIT {
+
+	@Autowired
+	private SnertDAO snertDAO;
+
+	@Autowired
+	private EntityManagerFactory entityManagerFactory;
+
+	@Before
+	public void checkJta() {
+		assertThat( entityManagerFactory.unwrap( SessionFactoryImplementor.class )
+				.getServiceRegistry().getService( TransactionCoordinatorBuilder.class ) )
+				.returns( true, TransactionCoordinatorBuilder::isJta );
+	}
+
+	@After
+	public void checkNoMemoryLeak() {
+		assertThat( HibernateOrmSearchSessionHolder.staticMapSize() ).isZero();
+	}
+
+	@Test
+	public void test() throws Exception {
+		Snert snert = new Snert();
+		snert.setId( 1L );
+		snert.setName( "dave" );
+		snert.setNickname( "dude" );
+		snert.setAge( 99 );
+		snert.setCool( Boolean.TRUE );
+		snertDAO.persist( snert );
+
+		await( "Waiting for indexing assertions" )
+				.untilAsserted( () -> assertThat( TimeoutFailureCollector.EXCEPTIONS ).isNotEmpty() );
+
+		Throwable exception = TimeoutFailureCollector.EXCEPTIONS.iterator().next();
+		Throwable cause = exception.getCause().getCause().getCause();
+		assertThat( cause ).hasMessageContaining( "transaction timed out" );
```

Comment:
I agree with the fact that a user usually uses this setting to increase the timeout so that the transactions will be executed. 
However, according to the [Bitronix documentation](https://github.com/bitronix/btm/wiki/Transaction-manager-configuration#bitronixtmtimerdefaulttransactiontimeout) the default timeout is 60 seconds.
Given that, I don't know if it is the case to introduce a sleep of 60 seconds to trigger a timeout using the default.
To be honest, I think that testing the opposite case, that is having a timeout setting the property to a lower value and not having a timeout with the default, would be the same in terms of verifying the fact that the property works. That is in my opinion what we really need to test here.
Doing that we would need a 1 second sleep that is better.